debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.*;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.*;
+import java.util.Date;
+import java.util.stream.Stream;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+public abstract class DebeziumTimestampConverter<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    public static final String OVERVIEW_DOC =
+            "Convert Debezium timestamps between different formats such as Unix epoch, strings, and Connect Date/Timestamp types."
+                    + "Applies to org.apache.kafka.connect.data.Struct fields or individual fields or to the entire value."
+                    + "<p/>Use the concrete transformation type designed for the record key (<code>" + DebeziumTimestampConverter.Key.class.getName() + "</code>) "
+                    + "or value (<code>" + DebeziumTimestampConverter.Value.class.getName() + "</code>).";
+
+    public static final String FIELD_CONFIG = "field";
+    private static final String FIELD_DEFAULT = "";
+
+    public static final String TARGET_TYPE_CONFIG = "target.type";
+
+    public static final String TIMESTAMP_FORMAT_CONFIG = "timestamp.format";
+    private static final String TIMESTAMP_FORMAT_DEFAULT = "yyyy-MM-dd HH:mm:ss";
+
+    public static final String DATE_FORMAT_CONFIG = "date.format";
+    private static final String DATE_FORMAT_DEFAULT = "yyyy-MM-dd";
+
+    public static final String FIELD_TYPE_CONFIG = "field.type";
+    private static final String FIELD_TYPE_DEFAULT = "empty.value";
+
+    public static final String STRUCT_FIELD_NAME_CONFIG = "struct.field";
+    private static final String STRUCT_FIELD_NAME_DEFAULT = "";
+
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef()
+            .define(FIELD_CONFIG, ConfigDef.Type.STRING, FIELD_DEFAULT, ConfigDef.Importance.LOW,
+                    "The field containing the timestamp, or empty if the entire value is a timestamp")
+            .define(TARGET_TYPE_CONFIG, ConfigDef.Type.STRING, ConfigDef.Importance.LOW,
+                    "The desired timestamp representation: string, unix, Date, Time, or Timestamp")
+            .define(TIMESTAMP_FORMAT_CONFIG, ConfigDef.Type.STRING, TIMESTAMP_FORMAT_DEFAULT, ConfigDef.Importance.LOW,
+                    "A SimpleDateFormat-compatible format for the timestamp. Used to generate the output when type=string "
+                            + "or used to parse the input if the input is a string.")
```

Comment:
How about putting the documentation as `private static final String FIELD_DOC = "...";` fields? I think that would make these `define(...)` lines much easier to follow.